lxd/api_cluster.go

```diff
@@ -1615,6 +1623,99 @@ func clusterNodeDelete(d *Daemon, r *http.Request) response.Response {
 	return response.EmptySyncResponse
 }
 
+// swagger:operation POST /1.0/cluster/certificates cluster clustering_update_certs
+//
+// Update certificates for the cluster
+//
+// Replaces existing cluster certificates and reloads LXD on each cluster member
+//
+// ---
+// consumes:
+//   - application/json
+// produces:
+//   - application/json
+// parameters:
+//   - in: body
+//     name: cluster
+//     description: Cluster certificate replace request
+//     required: true
+//     schema:
+//       $ref: "#/definitions/ClusterCertificatesPost"
+// responses:
+//   "200":
+//     $ref: "#/responses/EmptySyncResponse"
+//   "400":
+//     $ref: "#/responses/BadRequest"
+//   "403":
+//     $ref: "#/responses/Forbidden"
+//   "500":
+//     $ref: "#/responses/InternalServerError"
+func clusterCertificatesPost(d *Daemon, r *http.Request) response.Response {
+	req := api.ClusterCertificatesPost{}
+
+	// Parse the request
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	rawCert, err := base64.StdEncoding.DecodeString(req.ClusterCertificate)
+	if err != nil {
+		return response.BadRequest(fmt.Errorf("Certificate must be base64 encoded PEM certificate: %v", err))
+	}
+
+	rawKey, err := base64.StdEncoding.DecodeString(req.ClusterCertificateKey)
+	if err != nil {
+		return response.BadRequest(fmt.Errorf("Private key must be base64 encoded PEM key: %v", err))
+	}
+
```

Comment:
For validation as we're expecting pem. It may be best to use `encoding/pem` and `pem.Decode` instead?